src/StorageCache.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace BlackBox;
+
+use IteratorAggregate;
+
+/**
+ * Caches an storage with another storage.
+ *
+ * @author Carlos Lombarte <lombartec@gmail.com>
+ */
+class StorageCache implements IteratorAggregate, Storage
+{
+    /**
+     * The storage that will be cached.
+     *
+     * @var Storage
+     */
+    private $sourceStorage;
+
+    /**
+     * The cache used to storage the decorated storage.
+     *
+     * @var Storage
+     */
+    private $storageCache;
+
+    /**
+     * @param Storage $sourceStorage
+     * @param Storage $storageCache
+     */
+    public function __construct(Storage $sourceStorage, Storage $storageCache)
+    {
+        $this->sourceStorage    = $sourceStorage;
+        $this->storageCache     = $storageCache;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($id)
+    {
+        $cachedData = $this->storageCache->get($id);
+
+        if (null === $cachedData) {
+            return $this->sourceStorage->get($id);
```

Comment:
Here we get from the storage but we don't cache the result. So unless we manually put stuff in the storage nothing will be cached.

It should be rather something looking like this:

```php
data = $this->sourceStorage->get($id);
$this->storageCache->set($id, $data);
return $data;
```